show/main.py

```diff
@@ -557,12 +556,13 @@ def get_bgp_neighbor_ip_to_name(ip, static_neighbors, dynamic_neighbors):
 # This is our entrypoint - the main "show" command
 # TODO: Consider changing function name to 'show' for better understandability
 @click.group(cls=AliasedGroup, context_settings=CONTEXT_SETTINGS)
-def cli():
+@click.pass_context
+def cli(ctx):
     """SONiC command line - 'show' command"""
-    global config_db
 
-    config_db = ConfigDBConnector()
-    config_db.connect()
+    ctx.obj = Db()
+
+pass_db = click.make_pass_decorator(Db, ensure=True)
```

Comment:
this is the decorator. https://click.palletsprojects.com/en/7.x/complex/?highlight=make_pass_decorator